### PR TITLE
test: merge serial tests in cmd, planner, server, util

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: None

Problem Summary: After #30692, all the tests are run in serial. The names of *_serial_test.go are no longer appropriate.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: util/ranger/ranger_test.go [hunk 1]
```diff
@@ -1420,3 +1421,347 @@ func TestPrefixIndexAppendPointRanges(t *testing.T) {
...
 	}
 }
+
+func TestIndexRange(t *testing.T) {
+	dom, store, err := newDomainStoreWithBootstrap(t)
+	defer func() {
+		dom.Close()
+		require.NoError(t, store.Close())
+	}()
+	require.NoError(t, err)
```
Comment (hawkingrei) [inline util/ranger/ranger_test.go:1431]: L1431 move to L1427

**Comment by hawkingrei** [comment]
LGTM

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**